PrivateMaps/FaireWeatherTweakEx.py

```diff
@@ -5161,42 +5170,156 @@ def afterGeneration():
                     plot.setEurope(europeWest)
                 else:
                     plot.setEurope(europeEast)
-        
-    createIce()
+
     generateShallowCoast()
     generateShrubland()
     generateTaiga()
     generateRockSteppes()
     generateWetland()
+    createIce()
     mg = CyMapGenerator()
     mg.addFeaturesOnLand()
     mg.addFeaturesOnWater()
-    
+
+# both arguments must be floating point numbers between 0.0 and 1.0
+def addProbability(p1, p2):
+    return 1.0 - (1.0 - p1) * (1.0 - p2)
+
+# This Function creates Ice Caps on poles with slightly random penetration, depending on following factors:
+# Map Size, tile surrounding and how far from edge of the map tile actually is.
+# The resulting caps should form irregular shape, but still remain quite solid
 def createIce():
-    gc = CyGlobalContext()
-    mmap = gc.getMap()
-    featureIce = gc.getInfoTypeForString("FEATURE_ICE")
-    iceChance = 1.0 
-    # TAC - Map scripts - koma13 - START
-    #for y in range(4):
-    if abs(mc.bottomLattitude) == 90 and (mc.bottomLattitude != mc.topLattitude or mc.topLattitude < 0):
-       for y in range(4):
+    ## Those are magic numbers, empirically balanced:
+
+    # There is a very little chance that a tile can be ice-capped without corresponding surrounding
+    initialChance = 0.002
+    chanceIncreaseAdjacent = 0.35
+    chanceIncreaseDiagonal = 0.18
+    chanceReductionFromLatitude = 0.20 * mc.height ** 0.287
+
+    numOfIterations = 3
+
+    def isPlotOutOfBounds(x, y):
+        if x < 0 or x >= mc.width or y < 0 or y >= mc.height:
+            return True
+        else:
+            return False
+
+    def calculateIceChanceFromAdjacent(x, y):
+        # More chance for Ice if adjacent tiles are Ice, Permafrost or map bounds
+        totalChance = initialChance
+        for direction in range(1, 5):
```

Comment:
an alternate way to write it in a single loop would be :
```Python
      # this list will give the right chances depending on the directions, as they are well ordered
       chances =[0] + [chanceIncreaseAdjacent] *4 + [chanceIncreaseDiagonal] *4
      
       for direction in range(1, 9):
            xx, yy = GetXYFromDirection(x, y, direction)
            if isPlotWithinBounds(xx, yy) and (xx, yy) in icePlots:
                totalChance = addProbability(totalChance, chances[direction])
       return totalChance
```

Reply:
Applied your approach